src/bsp/platform/esp32/xi_bsp_io_fs_esp32.c

```diff
@@ -0,0 +1,307 @@
+    {
+        xi_bsp_debug_logger( "\nInvalid resource handle - Filesystem read failed" );
+        return XI_BSP_IO_FS_INVALID_PARAMETER;
+    }
+
+    if ( ( esp_ota_handle_t )resource_handle == open_firmware_bin_handle )
+    {
+        xi_bsp_debug_logger( "Error: No FS read of FW files in ESP32's OTA API" );
+        return XI_BSP_IO_FS_NOT_IMPLEMENTED;
+    }
+    else
+    {
+        /* make an allocation for memory block */
+        if ( NULL == *buffer )
+        {
+            *buffer = ( uint8_t* )xi_bsp_mem_alloc( xi_bsp_io_fs_buffer_size );
+            memset( ( uint8_t* )*buffer, 0, ( size_t )xi_bsp_io_fs_buffer_size );
+        }
+
+        /* let's set an offset */
+        fop_ret = fseek( fp, offset, SEEK_SET );
+
+        /* if error on fseek check errno */
+        XI_BSP_IO_FS_CHECK_CND(
+            fop_ret != 0, xi_bsp_io_fs_posix_errno_2_xi_bsp_io_fs_state( errno ), ret );
+
+        /* use the fread to read the file chunk */
+        fop_ret = fread( ( uint8_t* )*buffer, ( size_t )1,
+                         ( size_t )xi_bsp_io_fs_buffer_size, fp );
+
+        /* if error on fread check errno */
+        XI_BSP_IO_FS_CHECK_CND(
+            fop_ret == 0, xi_bsp_io_fs_posix_errno_2_xi_bsp_io_fs_state( errno ), ret );
+
+        *buffer_size = fop_ret;
+    }
+    return XI_BSP_IO_FS_STATE_OK;
+
+err_handling:
+    *buffer      = NULL;
+    *buffer_size = 0;
+    if ( NULL != *buffer )
+    {
+        xi_bsp_mem_free( ( uint8_t* )*buffer );
+    }
+
+    return ret;
+}
+
+xi_bsp_io_fs_state_t
+xi_bsp_io_fs_close( const xi_bsp_io_fs_resource_handle_t resource_handle )
+{
+    xi_bsp_io_fs_state_t ret = XI_BSP_IO_FS_STATE_OK;
+    FILE* fp                 = ( FILE* )resource_handle;
+    int fop_ret              = 0;
+
+    if ( ( esp_ota_handle_t )resource_handle == open_firmware_bin_handle )
+    {
+        esp_err_t retv = ESP_OK;
+
+        retv = esp_ota_end( resource_handle );
+        if ( ESP_OK != retv )
+        {
+            xi_bsp_debug_format( "esp_ota_end() failed with error %d", retv );
+            return XI_BSP_IO_FS_WRITE_ERROR;
+        }
+    }
+    else
+    {
+        if ( XI_BSP_IO_FS_INVALID_RESOURCE_HANDLE == resource_handle )
+        {
+            return XI_BSP_IO_FS_INVALID_PARAMETER;
+        }
+
+        fop_ret = fclose( fp );
+
+        /* if error on fclose check errno */
+        XI_BSP_IO_FS_CHECK_CND(
+            0 != fop_ret, xi_bsp_io_fs_posix_errno_2_xi_bsp_io_fs_state( errno ), ret );
+    }
+    return XI_BSP_IO_FS_STATE_OK;
+
+err_handling:
+    return ret;
+}
+
+xi_bsp_io_fs_state_t xi_bsp_io_fs_remove( const char* const resource_name )
+{
+    int ret = -1;
+    if ( !xi_bsp_fwu_is_this_firmware( resource_name ) )
+    {
+        xi_bsp_debug_logger( "FS Remove not implemented for FW images" );
+        return XI_BSP_IO_FS_NOT_IMPLEMENTED;
+    }
+    else
+    {
+        char filepath[64] = "";
+        snprintf( filepath, 64, "%s/%s", ESP32_FS_BASE_PATH, resource_name );
```

Comment:
is 64 enough?

Reply:
I think so. I'm gonna increase it to 128, just in case.